lumen/ai/embeddings.py

```diff
@@ -12,6 +13,15 @@
 STOP_WORDS_RE = re.compile(r"\b(?:{})\b".format("|".join(STOP_WORDS)), re.IGNORECASE)
 
 
+def _deterministic_hash(text):
+    """
+    Deterministic hash function using MD5.
+    Returns a stable integer hash that is consistent across Python sessions.
+    """
+    hash_bytes = hashlib.md5(text.encode('utf-8')).digest()
+    return int.from_bytes(hash_bytes[:4], byteorder='big')
```

Comment:
The undeterministic one was wildly varying, resulting in unstable vector store lookups, which made it really hard to debug.